picocrt/crt0.h

```diff
@@ -84,6 +84,13 @@ extern void __libc_init_array(void);
 #include <semihost.h>
 #endif
 
+/* Hexagon semihosting provides get_cmdline in its machine layer.
+ * Forward-declare it here to avoid including machine-specific headers.
+ */
+#ifdef __HEXAGON_ARCH__
+int get_cmdline(char *buffer, int count);
```

Comment:
It seems like you could use `sys_semihost_get_cmdline` for hexagon? That would avoid a bunch of the conditionals in this file, as well as duplicating the declaration of this symbol?